app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -77,26 +76,41 @@ export default function ExploreDisplayTabs({
   const [currentPointsSelected, setCurrentPointsSelected] = useState(null)
 
   // Differential expression settings
-  // TODO (SCP-4321): Try encapsulating these in a DE-specific component,
-  // to simplify this high-level ExploreDisplayTabs component
+  // TODO (SCP-4374): Integrate is_differential_expression_enabled from API
   const [showDeGroupPicker, setShowDeGroupPicker] = useState(false)
   const [deGenes, setDeGenes] = useState(null)
   const [deGroup, setDeGroup] = useState(null)
+  const [showDifferentialExpressionPanel, setShowDifferentialExpressionPanel] = useState(deGenes !== null)
 
-  // For readability, until approach outlined in TODO above can be fully
-  // attempted, which will likely be tightly couple with UX changes suggested
-  // at 2022-05-06 demo.
-  const showDifferentialExpressionPanel = deGenes !== null
+  // Hash of trace label names to the number of points in that trace
+  const [countsByLabel, setCountsByLabel] = useState(null)
 
-  // TODO (SCP-4321): In addition to feature flag, check
+  // TODO (SCP-4374): In addition to feature flag, check
   // is_differential_expression_enabled attribute from forthcoming update to
   // an API response
-  let isDifferentialExpressionEnabled = false
   const flags = getFeatureFlagsWithDefaults()
-  if (flags?.differential_expression_frontend) {
-    isDifferentialExpressionEnabled = true
+  const availableDeClusterAnnotations = [
```

Comment:
I'm assuming all this will be addressed in SCP-4374?

Reply:
Yup, it's one of the things entailed in TODO above.